junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/ClasspathResourceSelector.java

```diff
@@ -65,6 +83,32 @@ public String getClasspathResourceName() {
 		return this.classpathResourceName;
 	}
 
+	/**
+	 * Get the selected {@link Resource resources}.
+	 *
+	 * <p>If the {@link Resource resources} were not provided, but only their name,
+	 * this method attempts to lazily load the {@link Resource resources} based on
+	 * their name and throws a {@link PreconditionViolationException} if the
+	 * resource cannot be loaded.
+	 *
+	 * @since 1.12
+	 */
+	@API(status = EXPERIMENTAL, since = "1.12")
+	public Set<Resource> getClasspathResources() {
+		if (this.classpathResources == null) {
+			Try<Set<Resource>> tryToGetResource = ReflectionUtils.tryToGetResources(this.classpathResourceName);
+			Set<Resource> classpathResources = tryToGetResource.getOrThrow( //
+				cause -> new PreconditionViolationException( //
+					"Could not load resource(s) with name: " + this.classpathResourceName, cause));
+			if (classpathResources.isEmpty()) {
+				throw new PreconditionViolationException(
+					"Could not load resource(s) with name: " + this.classpathResourceName);
```

Comment:
I think it could be helpful to provide a different error message in this case. WDYT?

```suggestion
					"Could not find any resources with name: " + this.classpathResourceName);
```